edgarpecero/OMN-Notification-Service
# OMN-34:: created SES config file to initialize it and SES service with registerEmail and sendEmail endpoint

--

### Summary of Changes
- Added Amazon SES configuration file and SES service to initialize and manage email functionalities.
- Included Docker and Kubernetes manifest files for containerizing and deploying the application on AWS EKS.
- Configured GitHub Actions for continuous integration and deployment.

### Task References
- [OMN-34] Create SES config file, Dockerfile, Kubernetes manifest, and CI/CD pipeline.

### Detailed Changes
- Created `SESConfig` file to initialize SES client and handle email setup and configuration.
- Implemented `SESSendEmailService` to handle sending emails via SES, including validation and error handling.
- Added `Dockerfile` for containerizing the application, targeting an Amazon Corretto base image.
- Created Kubernetes manifest files to define deployment and service configurations for AWS EKS.
- Configured GitHub Actions workflow to automate building, testing, and deployment steps.

### Testing Instructions
1. Set up SES permissions and add verified email in AWS SES console for testing.
2. Test email sending functionality through the SES service by triggering sample endpoints.
3. Build and deploy the Docker container to verify successful packaging and startup.
4. Use Kubernetes manifests to deploy the application to AWS EKS and ensure proper scaling and networking.
5. Verify GitHub Actions by pushing a new commit to trigger CI/CD pipeline and monitor for any build or deployment errors.

### Future Considerations
- Explore options for handling high email volumes with SES, including potential integration with a queuing system.
- Add monitoring for SES metrics, such as email delivery rates and errors, to ensure reliability.
- Consider implementing IAM roles with least-privilege access for enhanced security.

---

### Additional Notes
- Ensure SES is in production mode to allow sending emails to unverified addresses.
- Review Kubernetes manifest files to confirm configuration settings align with EKS standards.
- Double-check GitHub Actions secrets for AWS credentials and permissions.
